### ISSUE TITLE
CF Push strips double dashes on route name in version 6.34.0

### ISSUE THREAD

**Issue by joshzarrabi** [cloudfoundry/cli#1307]
## Command  

`cf push my--app`

## What occurred

```
Pushing from manifest to org system / space system as admin...
Using manifest file some-path.yml
Getting app info...
Creating app with these attributes...
+ name:         my--app
  path:         some-app-path
+ buildpack:    go_buildpack
+ disk quota:   32M
+ memory:       32M
  env:
+   GOPACKAGENAME
  routes:
+   my-app.apps.example.com
```

## What you expected to occur

the route should be `my--app.apps.example.com`

## CLI Version  

6.34.0

**Comment by XenoPhex** [comment]
@joshzarrabi if you can't wait for a release, the fix is on our [edge build](https://github.com/cloudfoundry/cli/wiki/Edge-Binaries).

**Comment by joshzarrabi** [comment]
@XenoPhex we just pinned back for now. Thanks so much

**Comment by jbpivotal** [comment]
@joshzarrabi I'm curious...why do you use two dashes in an app name? What would be the impact if that wasn't an option?

**Comment by ryanmoran** [comment]
@jbpivotal it just happened to us by accident.

We have a set of acceptance tests that cover the networking functionality of Cloudfoundry. Those tests allow a developer to provide a "prefix" to include in front of the name of most resources provisioned during the test run (apps, orgs, spaces, etc.). This prefix makes it easy to identify those things that may be orphaned by the tests when they fail or don't cleanup correctly.

For our prefix, we chose a value like `test-`. This means that most resources are given names like `test-org-<GUID>` which is nice for a human to read and find. However, for some number of apps that are pushed to the platform during the test suite, the generated name ended up looking something like `test--app-<GUID>`. This was caused by some string interpolation that looked like the following `<prefix>-app-<GUID>`.

The expectation about whether a prefix could contain a trailing hyphen was changed in that one place. Unfortunately, this had been working fine for many versions of the `cf` CLI before 6.34.0, but broke when we upgraded to the latest version.

For us, the impact of this difference is that a test suite that we don't own, but that we use to validate the platform would break. We don't have a lot of control over how that suite functions. It would require the team that owns it to make some changes to support the new route naming scheme.
